src/main/java/address/sync/CloudSimulator.java

```diff
@@ -1,129 +1,670 @@
+     *
+     * @param personList
+     * @param newPerson
+     * @return newPerson, if added, else null
+     */
+    private CloudPerson addPerson(List<CloudPerson> personList, CloudPerson newPerson) throws JAXBException {
+        if (newPerson == null) throw new IllegalArgumentException("Person cannot be null");
+        String newPersonFirstName = newPerson.getFirstName();
+        String newPersonLastName = newPerson.getLastName();
+        if (newPersonFirstName == null || newPersonLastName == null) {
+            throw new IllegalArgumentException("Fields cannot be null");
+        }
+        if (isExistingPerson(personList, newPerson)) throw new IllegalArgumentException("Person already exists");
+
+        personList.add(newPerson);
+
+        return newPerson;
+    }
+
+    private Optional<CloudPerson> getPerson(List<CloudPerson> personList, String firstName, String lastName) {
+        return personList.stream()
+                .filter(person -> person.getFirstName().equals(firstName)
+                        && person.getLastName().equals(lastName))
+                .findAny();
+    }
+
+    private CloudPerson updatePersonDetails(List<CloudPerson> personList, String oldFirstName, String oldLastName,
+                                            CloudPerson updatedPerson) throws NoSuchElementException {
+        CloudPerson oldPerson = getPersonIfExists(personList, oldFirstName, oldLastName);
+        oldPerson.updatedBy(updatedPerson);
+        return oldPerson;
+    }
+
+    private CloudPerson getPersonIfExists(List<CloudPerson> personList, String oldFirstName, String oldLastName) {
+        Optional<CloudPerson> personQueryResult = getPerson(personList, oldFirstName, oldLastName);
+        if (!personQueryResult.isPresent()) throw new NoSuchElementException("No such person found.");
+
+        return personQueryResult.get();
+    }
+
+    private List<CloudPerson> mutateCloudPersonList(List<CloudPerson> cloudPersonList) {
+        modifyCloudPersonList(cloudPersonList);
+        addCloudPersonsBasedOnChance(cloudPersonList);
+        return cloudPersonList;
+    }
+
+    private List<CloudTag> mutateCloudTagList(List<CloudTag> cloudTagList) {
+        modifyCloudTagListBasedOnChance(cloudTagList);
+        addCloudTagsBasedOnChance(cloudTagList);
+        return cloudTagList;
+    }
+
+    private void modifyCloudPersonList(List<CloudPerson> cloudPersonList) {
+        cloudPersonList.stream()
+                .forEach(cloudPerson -> modifyCloudPersonBasedOnChance(cloudPerson));
+    }
+
+    private void modifyCloudTagListBasedOnChance(List<CloudTag> cloudTagList) {
+        cloudTagList.stream()
+                .forEach(cloudTag -> modifyCloudTagBasedOnChance(cloudTag));
+    }
+
+    private void addCloudPersonsBasedOnChance(List<CloudPerson> personList) {
+        for (int i = 0; i < MAX_NUM_PERSONS_TO_ADD; i++) {
+            if (RANDOM_GENERATOR.nextDouble() <= ADD_PERSON_PROBABILITY) {
```

Comment:
This should also check whether `shouldSimulateUnreliableNetwork` is true. Currently, it ALWAYS has a chance to mutate.